code/modules/materials/Mat_Materials.dm

```diff
@@ -257,13 +446,104 @@ ABSTRACT_TYPE(/datum/material)
 			X.execute(owner, attackobj, attacker, meleeorthrow)
 		return
 
+//Material definitions
 
+ABSTRACT_TYPE(/datum/material/interpolated)
 /datum/material/interpolated
 	mat_id = "imcoderium"
 	name = "imcoderium"
 	desc = "You should not be seeing this"
 	color = "#6f00ff"
 
+	New(var/datum/material/mat1,var/datum/material/mat2,var/t)
+		..()
+		var/ot = 1 - t
+		src.quality = round(mat1.quality * ot + mat2.quality * t)
+
+		src.prefixes = (mat1.prefixes | mat2.prefixes)
+		src.suffixes = (mat1.suffixes | mat2.suffixes)
+
+		src.value = round(mat1.value * ot + mat2.value * t)
+		src.name = mat1.interpolateName(mat2, 0.5)
+		src.desc = "This is an alloy of [mat1.name] and [mat2.name]"
+		src.mat_id = "([mat1.getID()]+[mat2.getID()])"
+		src.alpha = round(mat1.alpha * ot + mat2.alpha * t)
+		if(islist(mat1.color) || islist(mat2.color))
+			var/list/colA = normalize_color_to_matrix(mat1.color)
+			var/list/colB = normalize_color_to_matrix(mat2.color)
+			src.color = list()
+			for(var/i in 1 to length(colA))
+				src.color += colA[i] * ot + colB[i] * t
+		else
+			src.color = rgb(round(GetRedPart(mat1.color) * ot + GetRedPart(mat2.color) * t), round(GetGreenPart(mat1.color) * ot + GetGreenPart(mat2.color) * t), round(GetBluePart(mat1.color) * ot + GetBluePart(mat2.color) * t))
+		src.properties = mergeProperties(mat1.properties, mat2.properties, t)
+
+		src.edible_exact = round(mat1.edible_exact * ot + mat2.edible_exact * t)
+		if(src.edible_exact >= 0.5) src.edible = TRUE
+		else src.edible = FALSE
+
+		src.special_naming = FALSE // the naming proc doesn't carry over anyway
+
+		src.mixOnly = FALSE
+
+		//--
+		src.triggersTemp = getFusedTriggers(mat1.triggersTemp, mat2.triggersTemp, src)
+		src.triggersChem = getFusedTriggers(mat1.triggersChem, mat2.triggersChem, src)
+		src.triggersPickup = getFusedTriggers(mat1.triggersPickup, mat2.triggersPickup, src)
+		src.triggersDrop = getFusedTriggers(mat1.triggersDrop, mat2.triggersDrop, src)
+		src.triggersExp = getFusedTriggers(mat1.triggersExp, mat2.triggersExp, src)
+		src.triggersOnAdd = getFusedTriggers(mat1.triggersOnAdd, mat2.triggersOnAdd, src)
+		src.triggersOnLife = getFusedTriggers(mat1.triggersOnLife, mat2.triggersOnLife, src)
+		src.triggersOnAttack = getFusedTriggers(mat1.triggersOnAttack, mat2.triggersOnAttack, src)
+		src.triggersOnAttacked = getFusedTriggers(mat1.triggersOnAttacked, mat2.triggersOnAttacked, src)
+		src.triggersOnEntered = getFusedTriggers(mat1.triggersOnEntered, mat2.triggersOnEntered, src)
+
+		handleTriggerGenerations(src.triggersTemp)
+		handleTriggerGenerations(src.triggersChem)
+		handleTriggerGenerations(src.triggersPickup)
+		handleTriggerGenerations(src.triggersDrop)
+		handleTriggerGenerations(src.triggersExp)
+		handleTriggerGenerations(src.triggersOnAdd)
+		handleTriggerGenerations(src.triggersOnLife)
+		handleTriggerGenerations(src.triggersOnAttack)
+		handleTriggerGenerations(src.triggersOnAttacked)
+		handleTriggerGenerations(src.triggersOnEntered)
```

Comment:
would also make this look less like an unrolled loop

Reply:
I made a change, now it looks less like an unrolled loop. I'm not sure it's better though.